jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/KModuleDeploymentService.java

```diff
@@ -176,7 +176,7 @@ public void deploy(DeploymentUnit unit) {
     	        	processResources(depModule, files, kieContainer, kmoduleUnit, deployedUnit, depModule.getReleaseId(), processDescriptors);
     	        }
             }
-            Collection<ReleaseId> dependencies = module.getJarDependencies(new DependencyFilter.ExcludeScopeFilter("test"));
+            Collection<ReleaseId> dependencies = module.getJarDependencies(new DependencyFilter.ExcludeScopeFilter("test", "provided"));
```

Comment:
it was overlooked from the time when it was added, as it should always exclude test and provided scope. I discovered it while working on this when you define dependency to jbpm/drools dependencies e.g. jbpm-kie-services or jbpm-services-api to implement custom mappers or query param builders. Then it scans and attaches jbpm-kie-services and all its transitive deps into project class loader ... which is time consuming and unnecessary so decided to fix it here. Can be changed as separate commit too if we think that will be better from visibility point of view